raiden/transfer/architecture.py

```diff
@@ -138,3 +138,23 @@ def dispatch(self, state_change):
         assert all(isinstance(e, Event) for e in events)
 
         return events
+
+    def __eq__(self, other):
+        if isinstance(other, StateManager):
```

Comment:
So, by "complex" I tried to mean something along the lines "I need to spend thought effort to understand", a new function in a different module to hide the introspective approach doesn't make things simpler IMO.

Why not just set the attributes to None in the constructor? Or use a `getattr` that defaults to None?